### DISCUSSION TITLE
Creating Custom Sum Field for Advanced Block

### DISCUSSION THREAD

**Discussion by Cattrah** [YetiForceCompany/YetiForceCRM#15814] [How to]
I want to create a new field (called Hours) for the advanced block in a module that will simply sum all the values in that column. I copied the code for the Quantity field and it renders properly in the advanced block, when it is saved, the values are summed properly in the Details view. The only problem I have yet to solve is in Edit view, the javascript is not updating the Summary of the column in the footer when there is a change in the field.
![image](https://user-images.githubusercontent.com/101741600/160032147-dc945ec3-9f84-4c27-8339-1b7b7fe2e84b.png)

The values are properly saved and summed in the details view
![image](https://user-images.githubusercontent.com/101741600/160032277-22e9c8b6-6b16-4b29-b230-05a707fb6860.png)

I have created a setHours() function in (basedir)/public_html/layouts/basic/modules/Vtiger/resources/Inventory.js and referenced it in the setRowData() function of the same file. I also registered the column class (.hrs) in registerRowChangeEvent() of this .js file too. Something is still not wired up correctly to fire the summation or record it to the summary footer. Not sure which, just looking for some direction on what file/function I should be looking into.

**Comment by mariuszkrzaczkowski** [comment]
I made a minor change that will easily allow you to do the conversion without any changes to the core
https://github.com/YetiForceCompany/YetiForceCRM/commit/1168eb8e243575e18af26ccf91a4229a48d628de#diff-8f8456f4033f5be2e4fd87ad8155556ff059e48f9b2886f9b8bc2ab72d1b5765R517-R519

**Reply by Cattrah** [reply]
Thanks, I applied this change, but the new Hours column is still not automatically summing in Edit View. Columns with Quantity, unit price, and net price are still operating normally.

**Reply by mariuszkrzaczkowski** [reply]
This is just a hook so as not to modify the engine
```JavaScript
  app.event.on('Inventory.RowCalculations', (e, self, row) => {
  
  });
```
you need to write a function to sum values
https://github.com/YetiForceCompany/YetiForceCRM/blob/bc646a754ab442065826ddaeb5ae0c48d10db3c0/public_html/layouts/basic/modules/Vtiger/resources/Inventory.js#L551-L562

**Comment by newminder** [comment]
did you fixed it? can you share the method?
